pkg/scheduler/service/transition_test.go

```diff
@@ -44,99 +41,123 @@ func TestTransition(t *testing.T) {
 	//create transition which will change cluster states
 	transition := newClusterStatusTransition(dbConn, inventory, reconRepo, logger.NewLogger(true))
 
-	testSchedulerConfig := &SchedulerConfig{
+	err = transition.StartReconciliation(clusterState.Cluster.RuntimeID, clusterState.Configuration.Version, &SchedulerConfig{
 		PreComponents:  nil,
 		DeleteStrategy: "",
-	}
+	})
+	require.NoError(t, err)
 
 	//cleanup at the end of the execution
-	defer func() {
+	cleanupFn := func() {
```

Comment:
I think you can simply use AfterTest for this instead of using a cleanup function

Reply:
This was not generic enough amongst tests of this package. For more generalisation, the tests need to be completely reworked on, which might go out of scope of the current PR. I've added an AfterTest(), which only serves this specific test for now.